packages/grafana-data/src/transformations/transformers/filterByName.ts

```diff
@@ -20,25 +21,38 @@ export const filterFieldsByNameTransformer: DataTransformerInfo<FilterFieldsByNa
    * Return a modified copy of the series. If the transform is not or should not
    * be applied, just return the input series
    */
-  operator: (options, replace) => (source) =>
+  operator: (options, ctx) => (source) =>
     source.pipe(
       filterFieldsTransformer.operator(
         {
-          include: getMatcherConfig(options.include),
-          exclude: getMatcherConfig(options.exclude),
+          include: getMatcherConfig(ctx, options.include, options.byVariable),
+          exclude: getMatcherConfig(ctx, options.exclude, options.byVariable),
         },
-        replace
+        ctx
       )
     ),
 };
 
 // Exported to share with other implementations, but not exported to `@grafana/data`
-export const getMatcherConfig = (options?: RegexpOrNamesMatcherOptions): MatcherConfig | undefined => {
+export const getMatcherConfig = (
+  ctx: DataTransformContext,
+  options?: RegexpOrNamesMatcherOptions,
+  byVariable?: boolean
+): MatcherConfig | undefined => {
   if (!options) {
     return undefined;
   }
 
-  const { names, pattern } = options;
+  const { names, pattern, variable } = options;
+
+  if (byVariable && variable) {
+    const stringOfNames = ctx.interpolate(variable);
+    if (/\{.*\}/.test(stringOfNames)) {
+      const n = stringOfNames.slice(1).slice(0, -1).split(',');
+      return { id: FieldMatcherID.byNames, options: { names: n } };
+    }
+    return { id: FieldMatcherID.byNames, options: { names: stringOfNames.split(',') } };
```

Comment:
Do we need the `split(',')` at the end here? Seems that this path will only result in single option values from my understanding.

Reply:
This path is if you have a variable where the value is a comma separated list of names. Maybe not the most useful scenario as it can only really happen if the user inputs it manually :thinking:

Reply:
I understand, thanks! Not sure if we'd need some more explanatory docs around this, as it seems to me a bit like a hidden feature, but I'll leave that decision to you.